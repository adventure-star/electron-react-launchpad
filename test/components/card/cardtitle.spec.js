import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CardTitle from '../../../src/renderer/components/card/cardtitle.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<CardTitle>Test title</CardTitle>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Card - Card title component', () => {
    it('should be of type span', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('span');
    });

    it('should have only the class "card-title"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card-title');
    });

    it('children should be the text "Test title"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test title');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
