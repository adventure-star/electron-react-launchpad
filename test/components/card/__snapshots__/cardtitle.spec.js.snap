// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card - Card title component Matches/Renders a snapshot 1`] = `
<span
  className="card-title"
>
  Test title
</span>
`;
